### PR TITLE
MNT: Added FBs for LIC&ATM following WFS

### PR DESCRIPTION
**Pull request by jsheppard95**
Added function blocks for LIC and ATM CCs which will be first instantiated in `lcls-plc-tmo-motion`:

- https://github.com/jsheppard95/lcls-plc-tmo-motion/blob/H1.1-CCs/plc-tmo-motion/tmo_motion/POUs/PRG_LI1K4_IP1.TcPOU
- https://github.com/jsheppard95/lcls-plc-tmo-motion/blob/H1.1-CCs/plc-tmo-motion/tmo_motion/POUs/PRG_TM1K4.TcPOU

Followed WFS as guide/template.

Will also use this PR to track changes that come up through checkouts.

TODO:
- Need to confirm state motion parameters: velocity, acceleration, etc.
- Need to confirm state enums:
  - LIC: Unknown, Out, In?
  - ATM: Unknown, Out, Target 1-5?

### CODE REVIEW

File: lcls2-cc-lib/Library/Devices/LIC/FB_LIC.TcPOU [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    nUnknownAssertionID: UDINT;
+END_VAR
+VAR
+    fbYStage: FB_MotionStage;
+
+    {attribute 'pytmc' := '
+        pv: MMS:STATE
+        io: i
+    '}
+    fbStates: FB_WFS_States;
```
Comment (ZLLentz) [inline lcls2-cc-lib/Library/Devices/LIC/FB_LIC.TcPOU:23]:
```suggestion
    fbStates: FB_LIC_States;
```

File: lcls2-cc-lib/Library/Devices/LIC/FB_LIC_States.TcPOU [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    bLICInit := TRUE;
+
+    stOut.sName := 'OUT';
+    stOut.fVelocity := fOutVelocity;
+    stOut.fDelta := fOutDelta;
+    stOut.fAccel := fAccel;
+    stOut.fDecel := fOutDecel;
+    stOut.bMoveOk := TRUE;
+
+    stTarget1.sName := 'TARGET1';
```
Comment (ZLLentz) [inline lcls2-cc-lib/Library/Devices/LIC/FB_LIC_States.TcPOU:47]: Should double check if they want this called `TARGET1`, `IN`, `MIRROR`, etc.

File: lcls2-cc-lib/Library/Library.plcproj [hunk 3]
```diff
@@ -137,13 +153,15 @@
...
     </PlaceholderResolution>
-    <PlaceholderResolution Include="lcls-twincat-motion">
-      <Resolution>lcls-twincat-motion, * (SLAC)</Resolution>
-    </PlaceholderResolution>
     <PlaceholderResolution Include="PMPS">
       <Resolution>PMPS, * (SLAC - LCLS)</Resolution>
     </PlaceholderResolution>
   </ItemGroup>
+  <ItemGroup>
+    <LibraryReference Include="lcls-twincat-motion,1.3.2,SLAC">
```
Comment (ZLLentz) [inline lcls2-cc-lib/Library/Library.plcproj:161]: Can't merge into lib if you pin the version. Libraries need to stay unpinned to the project that uses the library can decide the version.

File: lcls2-cc-lib/Library/Devices/LIC/FB_LIC.TcPOU (hunk 1 above)
Comment (jsheppard95) [reply lcls2-cc-lib/Library/Devices/LIC/FB_LIC.TcPOU]: Good catch, I knew I was gonna miss one. Resolved in commit 739e16d

File: lcls2-cc-lib/Library/Devices/LIC/FB_LIC_States.TcPOU (hunk 2 above)
Comment (jsheppard95) [reply lcls2-cc-lib/Library/Devices/LIC/FB_LIC_States.TcPOU:47]: Good call, I'll double check this with Justin, along with the state names for the ATM as this was also just a copy from the WFS.

File: lcls2-cc-lib/Library/Library.plcproj (hunk 3 above)
Comment (jsheppard95) [reply lcls2-cc-lib/Library/Library.plcproj]: Makes sense, good to know for libraries within libraries. Resolved in commit 739e16d

**Comment by jsheppard95** [comment]
Also it looks like the Travis build keeps failing at `TwinCAT XML Formatter....................................................Failed`. Does this indicate that my TwinCAT settings are not quite right?

**Comment by ZLLentz** [comment]
Yes, that is exactly right.
In the end though, only the linting/docs steps are important.

**Comment by ZLLentz** [comment]
According to the traveler, the LIC actually has 4 states:

"All Mirrors Out" at 0.118
"Bottom Mirror In" at -36.38
"Top Mirror In" at -70.38
"Scattering Target" at -102.38
(positions from LI1K4)

I will push an update to this PR to reflect the information in the traveler

**Comment by ZLLentz** [comment]
I'm going to finish and merge this one so TMO can run with a tagged library

**Review by ZLLentz — APPROVED** [review]

**Pull request merged**